Makefile

```diff
@@ -1,52 +1,191 @@
-DEFAULT_GOARCH := $(shell go env GOARCH)
-BUILD_HASH = $(shell git rev-parse HEAD)
-LDFLAGS += -X "github.com/mattermost/rtcd/service.buildHash=$(BUILD_HASH)"
-
-## Check go mod files consistency
-.PHONY: gomod-check
-gomod-check:
-	@echo Checking go mod files consistency
-	go mod tidy -v && git --no-pager diff --exit-code go.mod go.sum || (echo "Please run \"go mod tidy\" and commit the changes in go.mod and go.sum." && exit 1)
-
-.PHONY: check-style
-check-style: golangci-lint gomod-check
-	@echo Checking for style guide compliance
-
-.PHONY: golangci-lint
-golangci-lint: ## Run golangci-lint on codebase
-# https://stackoverflow.com/a/677212/1027058 (check if a command exists or not)
-	@if ! [ -x "$$(command -v golangci-lint)" ]; then \
-		echo "golangci-lint is not installed. Please see https://github.com/golangci/golangci-lint#install for installation instructions."; \
-		exit 1; \
-	fi; \
-
-	@echo Running golangci-lint
-	golangci-lint run ./...
 
-.PHONY: test
-test:
-	go test -v -mod=readonly -failfast -race ./...
+# Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+# See LICENSE for license information.
+
+# ====================================================================================
+# Variables
+# App Go Code location
+CONFIG_APP_CODE        += ./cmd/rtcd
+# Dockerfile's location
+CONFIG_DOCKERFILE      += ./build/Dockerfile
+# Docker options to inherit for all docker run commands
+CONFIG_DOCKEROPTS      += --rm -i -u $$(id -u):$$(id -g) --platform "linux/amd64"
+# Registry to upload images
+CONFIG_IMAGE_REGISTRY  += my.registy.com
+CONFIG_DOCKER_REPO     += mattermost
+# Temporary folder to output compiled binaries artifacts
+CONFIG_BIN_DIR         := dist
+## Docker Images Variables
+IMAGE_GOLINT           += "golangci/golangci-lint:v1.45.2@sha256:e84b639c061c8888be91939c78dae9b1525359954e405ab0d9868a46861bd21b"
+IMAGE_DOCKERLINT       += "hadolint/hadolint:v2.9.2@sha256:d355bd7df747a0f124f3b5e7b21e9dafd0cb19732a276f901f0fdee243ec1f3b"
+
+## Go Variables
+# Go executable
+GO         = go
+# LDFLAGS
+GO_LDFLAGS += -X "github.com/mattermost/${APP_NAME}/service.buildHash=$(APP_COMMIT)"
```

Comment:
ideally we want along with @streamer45 to add `buildVersion` and `buildDate`

Reply:
👍 
Will also expose `buildVersion` & `buildDate`

Reply:
Rebased and pushed changes